XenAdmin/Diagnostics/Checks/AssertCanEvacuateCheck.cs

```diff
@@ -78,8 +80,18 @@ protected List<Problem> CheckHost()
                     problems.Add(new LocalCD(this, residentVM));
                     VMsWithProblems.Add(residentVM.opaque_ref);
                 }
+
+                if (restrictMigration && residentVM.is_a_real_vm && !VMsWithProblems.Contains(residentVM.opaque_ref))

```

Comment:
I think we still care about the other problems, even when migration is restricted: We require to switch off VM's auto power, so that VMs don't start automatically while in the middle of an upgrade. And we require to eject the tools ISO before RPU, because after the upgrade, the old tools iso is guaranteed to be missing.